jep/0000/README.adoc

```diff
@@ -0,0 +1,325 @@
+discuss potential security issues and how they are mitigated or prevented,
+and detail how the JEP interacts with existing elements in Jenkins, such as permissions, authentication, authorization, etc.
+
+If this proposal will have no impact on security, this section may simply say:
+There are no security risks related to this proposal.
+====
+The intent of this proposal is to improve overall security for Jenkins instances that are using some form of matrix authorization.
+All configuration items that require `Jenkins.ADMINISTER`, `Jenkins.RUN_SCRIPTS`, `PluginManager.CONFIGURE_UPDATECENTER`, and `PluginManager.UPLOAD_PLUGINS`  are being reviewed to determine if they can be changed to require the new `Jenkins.CONFIGURE` permission type.
+The greatest risk is that some configuration may be _more restrictive_ than necessary, either because it was overlooked or due to differences in opinion during the review process.
+
+== Infrastructure Requirements
+
+There are no new infrastructure requirements related to this proposal.
+
+== Testing
+
+[TIP]
+====
+If the JEP involves any kind of behavioral change to code
+(whether in a Jenkins product or backend infrastructure),
+give a summary of how its correctness (and, if applicable, compatibility, security, etc.) can be tested.
+
+In the preferred case that automated tests can be developed to cover all significant changes, simply give a short summary of the nature of these tests.
+
+If some or all of the changes will require human interaction to verify them, explain why automated tests are considered impractical.
+Then, summarize what kinds of test cases might be required: user scenarios with action steps and expected outcomes.
+Detail whether behavior might be different based on the platform (operating system, servlet container, web browser, etc.)?
+Are there foreseeable interactions between different permissible versions of components (Jenkins core, plugins, etc.)?
+Does this change require that any special tools, proprietary software, or online service accounts to exercise a related code path (e.g., Active Directory server, GitHub login, etc.)?
+When will you complete testing relative to merging code changes, and might retesting be required if other changes are made to this area in the future?
+
+If this proposal requires no testing, this section may simply say:
+There are no testing issues related to this proposal.
+====
+Existing tests that validate permissions, or make assumptions about the permissions being used, will be updated to conform to the new permission model.
+Additional tests will be written that validate the new permission type cannot be used when the more restrictive `Jenkins.ADMINISTER` is needed.
+
+== Prototype Implementation
+[NOTE]
+The prototype code found below will be moved to a proper _DRAFT_ pull request in the near future
+
+A WIP prototype implementation can be found at:
+
+* link:https://github.com/mikecirioli/jenkins/tree/FGP[WIP branch with proposed changes]
```

Comment:
```suggestion
* link:https://github.com/jenkinsci/jenkins/compare/master...mikecirioli:FGP[WIP branch with proposed changes]
```

(though a draft PR would be even better)

Reply:
We will create a `Draft` PR in the next couple of days

Reply:
the draft PR is here https://github.com/jenkinsci/jenkins/pull/4374